lib/cunit/cu_pc_sort.c

```diff
@@ -0,0 +1,390 @@
+test_sort_patch_is_sorted_compression_dimensional_none()
+{
+    test_sort_patch_is_sorted_compression_dimensional(PC_DIM_NONE);
+}
+
+static void
+test_sort_patch_is_sorted_compression_dimensional_zlib()
+{
+    test_sort_patch_is_sorted_compression_dimensional(PC_DIM_ZLIB);
+}
+
+static void
+test_sort_patch_is_sorted_compression_dimensional_rle()
+{
+    test_sort_patch_is_sorted_compression_dimensional(PC_DIM_RLE);
+}
+
+static void
+test_sort_patch_is_sorted_compression_dimensional_sigbits()
+{
+    test_sort_patch_is_sorted_compression_dimensional(PC_DIM_SIGBITS);
+}
+
+static void
+test_sort_patch_ndims()
+{
+	// 00 endian (big)
+	// 00000000 pcid
+	// 00000000 compression
+	// 00000002 npoints
+	// 0000000800000001000000050006 pt1 (XYZi)
+	// 0000000200000003000000040008 pt2 (XYZi)
+	// 0000000200000002000000040008 pt2 (XYZi)
+
+    // init data
+    PCPATCH *pasort1, *pasort2;
+	char *hexbuf = "00000000000000000000000003000000080000000400000005000600000002000000030000000400080000000200000002000000040009";
+	size_t hexsize = strlen(hexbuf);
+	uint8_t *wkb = bytes_from_hexbytes(hexbuf, hexsize);
+	PCPATCH *pa = pc_patch_from_wkb(schema, wkb, hexsize/2);
+    const char *X[] = {"X"};
+    const char *Y[] = {"Y"};
+    const char *X_Y[] = {"X", "Y"};
+
+    // test that initial data is not sorted
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pa, X, 1, PC_FALSE), PC_FALSE);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pa, Y, 1, PC_FALSE), PC_FALSE);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pa, X_Y, 2, PC_FALSE), PC_FALSE);
+
+    // sort on X attribute and test
+    pasort1 = pc_patch_sort(pa, X, 1);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pasort1, X, 1, PC_TRUE), PC_TRUE);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pasort1, Y, 1, PC_TRUE), PC_FALSE);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pasort1, X_Y, 2, PC_TRUE), PC_FALSE);
+
+    // sort on X and Y and tst
+    pasort2 = pc_patch_sort(pa, X_Y, 2);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pasort2, X, 1, PC_TRUE), PC_TRUE);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pasort2, Y, 1, PC_TRUE), PC_TRUE);
+    CU_ASSERT_EQUAL(pc_patch_is_sorted(pasort2, X_Y, 2, PC_TRUE), PC_TRUE);
+
+    // free
+    pcfree(wkb);
+    pc_patch_free(pasort1);
+    pc_patch_free(pasort2);
+    pc_patch_free(pa);
+}
+
+/* REGISTER ***********************************************************/
+
+CU_TestInfo sort_tests[] = {
+	PC_TEST(test_sort_simple),
+    PC_TEST(test_sort_consistency),
+    PC_TEST(test_sort_one_point),
+    PC_TEST(test_sort_stable),
+    PC_TEST(test_sort_patch_is_sorted_no_compression),
+    PC_TEST(test_sort_patch_is_sorted_compression_dimensional_none),
+    PC_TEST(test_sort_patch_is_sorted_compression_dimensional_zlib),
+    PC_TEST(test_sort_patch_is_sorted_compression_dimensional_sigbits),
+    PC_TEST(test_sort_patch_is_sorted_compression_dimensional_rle),
+    PC_TEST(test_sort_patch_ndims),
+    CU_TEST_INFO_NULL
+};
+
+CU_SuiteInfo sort_suite = {"sort", init_suite, clean_suite, sort_tests};
```

Comment:
This will cause a segfault with cunit-2.3-1. See https://github.com/pgpointcloud/pointcloud/pull/124. I'll create a PR for that.

I'll also fix the indent in this file.